tests/unit/socket_test.cc

```diff
@@ -224,3 +224,29 @@ SEASTAR_TEST_CASE(socket_on_close_local_shutdown_test) {
         when_all(std::move(client), std::move(server)).discard_result().get();
     });
 }
+
+SEASTAR_TEST_CASE(socket_connect_abort_test) {
+    return seastar::async([&] {
+        bool too_late = false;
+        auto sk = make_socket();
+        auto cf = sk.connect(ipv4_addr("10.255.255.1", 12345)).then([] (auto cs) {
+            fmt::print("Connected\n");
+            BOOST_REQUIRE(false);
+        }).handle_exception([&too_late] (auto ex) {
+            fmt::print("Cannot connect {}\n", ex);
+            BOOST_REQUIRE(!too_late);
+        });
+
+        auto check = sleep(std::chrono::seconds(2)).then([&too_late] {
+            fmt::print("Connection must have been aborted already\n");
+            too_late = true;
```

Comment:
Oh, I think I understand now - it's not about timeout at all. The code below after 0.5 seconds calls shutdown()  on the connection, and then the connect() code (above)  will stop and check too_late wasn't yet set. And it wasn't - it will be set only after 2 seconds.

Maybe the code could have been organized in a different order to make this clearer, but at least it looks corect.

Reply:
Yes, it's exactly as you described.